shiksha-api/app-service/app/services/question_paper_service.py

```
@@ -441,15 +478,30 @@ async def _generate_questions_batch(
                     curr_message=user_message + "\n\nResponse format must conform to JSON schema:\n" + json.dumps(GeneratedQuestionItemResponse.model_json_schema()),
-                    chat_history=chat_history
+                    chat_history=chat_history,
                 )
                 # chat_with_index returns {"response": str, "source_nodes": list}
-                response_content = response_content["response"]
-                content = response_content.strip("```json").strip("```")
-                return GeneratedQuestionItemResponse.model_validate_json(content).items
+                response_content = rag_result["response"] if isinstance(rag_result, dict) else str(rag_result)
+
+            # Clean response
+            content = response_content.strip("```json").strip("```")
```

Ruff (0.15.13)

[warning] Using `.strip()` with multi-character strings is misleading

(B005)